### ISSUE TITLE
`object-curly-spacing` and object destructuring in function parameters with flow types

### ISSUE THREAD

**Issue by hzoo** [babel/eslint-plugin-babel#18]
```js
type Data = {
  type: string;
  payload: Object
};

export function test({type, payload}: Data) {
  console.log(type, payload);
}
```

`There should be no space before 'Data'`

Ref https://github.com/babel/babel-eslint/issues/185

**Comment by zaygraveyard** [comment]
And all tests from PR #40 pass so the problem might be resolved.
Can anyone check?

**Comment by sstur** [comment]
As far as I can tell, this problem is not resolved. This false positive still occurs in eslint@3.0.1 + babel-eslint@6.1.2 + eslint-plugin-babel@3.3.0

<img width="912" alt="screenshot 2016-07-14 15 40 20" src="https://cloud.githubusercontent.com/assets/369384/16833512/dbdafa5c-49d9-11e6-8f79-f5563c57f1c9.png">
<img width="629" alt="screenshot 2016-07-14 15 43 29" src="https://cloud.githubusercontent.com/assets/369384/16833513/dbe34a54-49d9-11e6-9c2c-5658f961d07f.png">

**Comment by zaygraveyard** [comment]
@sstur Thank you, I'll check it this false positive ASAP.

**Comment by zaygraveyard** [comment]
@sstur I was unable to reproduce this case I used `eslint@3.1.0` + `babel-eslint@6.1.2` + `eslint-plugin-babel@3.3.0`.
Can you provide a full example (full source code with the corresponding `.eslintrc`) that will reproduce this issue?
Thank you

**Comment by sstur** [comment]
Sure, will do! I'll try to come up with the simplest code required to repro.

**Comment by sstur** [comment]
OK, here goes:
https://github.com/sstur/example/blob/master/example.js

If you clone that repo, `npm install` and then `npm run lint` it should illustrate it perfectly.

Let me know if you need anything else..

**Comment by zaygraveyard** [comment]
Thank you.
It seems you are activating the wrong rule `object-curly-spacing` is the eslint rule and `babel/object-curly-spacing` is the eslint-plugin-babel one.
Let me know if this fixes your issue.

**Comment by sstur** [comment]
Yes! This fixes it. Thanks!

The part I missed is obvious in the readme:

> enable all the rules you would like to use (remember to disable the original ones as well!)

I appreciate your help.
